File: Java/p_002.java
class p_002 {
  static double fib(int n) {
    double x = 0;
    double y = 1;
    double tmp;
    for (int i = 0; i < n; i++) {
      tmp = x;
      x = y;
      y += tmp;
    }
    return y;
  }

  public static void main(String[] args) {
    double[] arr = new double[1000];
    for (int i = 1; i < arr.length; i++) {
      arr[i] = fib(i);
    }

    double total = 0;
    for (double el : arr) {
      if (el % 2 == 0) {
        total += el;
        // System.out.println(el);
      }
    }

    System.out.println(total);
  }
}

File: Python/p002.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
from time import time


def fib(n):
    x = 0  # Represents the current Fibonacci number being processed
    y = 1  # Represents the next Fibonacci number in the sequence
    for _ in range(n):
        x, y = y, x+y
    return y


start = time()
lsFib = (fib(x) for x in range(10000))

total = 0
for i in lsFib:
    if (i % 2 == 0):
        total += i

print(total)
print(time() - start)
